### PR TITLE
Factor out common model code to Model base class.

### PR DESCRIPTION
**Pull request by danieldk**
The placeholders, classification layers, loss, and accuracy computations
are shared between models. Add a Model base class with methods to create
these ops and use then in the rnn and convolution models.

This change also adds a CRF classification layer, since is was easy to
add with this refactoring. Unfortunately, in pure topological field
labling the use of CRFs does not improve accuracy.

### CODE REVIEW

File: toponn-utils/tensorflow/model.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        # Mask inactive timesteps
+        correct = tf.multiply(self.mask, tf.cast(correct, tf.float32))
+
+        # Compensate for inactive time steps.
+        correct = tf.reshape(correct, [-1])
+        correct = tf.truediv(correct, tf.reduce_mean(self.mask))
+
+        return tf.reduce_mean(correct, name="%s_accuracy" % prefix)
+
+    def affine_transform(self, prefix, x, n_outputs):
```
Comment (twuebi) [inline toponn-utils/tensorflow/model.py]: Is there a difference in performance to tf.layers.dense(n_outputs, use_bias=True, activation=None)?

**Review by twuebi — APPROVED** [review]

File: toponn-utils/tensorflow/model.py (hunk 1 above)
Comment (danieldk) [reply toponn-utils/tensorflow/model.py]: Looking at the implementation of the dense layer, it should boil down to the same. But since it's better to use canonical TF functions, I'll update this to use `dense`. Thanks for pointing this our!
